kratos/python/add_processes_to_python.cpp

```diff
@@ -148,6 +149,10 @@ void  AddProcessesToPython()
             init<ModelPart&, boost::numeric::ublas::vector<int> ,boost::numeric::ublas::vector<int> ,unsigned int>())
     ;
 
+    class_<StructuredMeshGeneratorProcess, bases<Process>, boost::noncopyable >("StructuredMeshGeneratorProcess",
+            init<Geometry< Node<3> >::Pointer, ModelPart&, Parameters&>())
```

Comment:
This will lead to a memory error!! Because inside the process you are storing a reference not the `Geotmetry::Pointer`  and if you remove the input geometry inside python before running the process `Execute ` method, it will crash!! you need to pass the const reference.

Reply:
To pass the constant you should say the lifetime of the constant with an additional argument: `with_custodian_and_ward<1,2>` which means that hold the argument 1 until this object is alive!

Reply:
Great! I have tested it by doing del to the geometry before Execure(). Indeed, without the suggested fix you run into a segmentation fault. With the fix it works without problem.